File: aluguel-core/src/main/java/com/nortetec/imovel/ImovelServiceImpl.java
package com.nortetec.imovel;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ImovelServiceImpl implements ImovelService {
    private final ImovelRepository repository;

    public ImovelServiceImpl(ImovelRepository repository) {
        this.repository = repository;
    }

    @Override
    public List<ImovelImpl> listaImovel() {
        return repository.findAll();
    }
}

File: src/main/kotlin/com/nortetec/imovel/ImovelServiceImpl.kt
package com.nortetec.imovel

import org.springframework.stereotype.Service

@Service
class ImovelServiceImpl(val repository: ImovelRepository) : ImovelService {
    override fun listaImovel(): List<Imovel> {
        return repository.findAll()
    }

    override fun salvarImovel(imovel: Imovel): String {
        repository.save(imovel)
        return "Endereco Salvo"
    }
}